acceptance/builder_test.go

```diff
@@ -0,0 +1,415 @@
+	)
+	defer h.DockerImageRemove(t, builderImage)
+
+	spec.Run(t, "acceptance-builder", testBuilder, spec.Parallel(), spec.Report(report.Terminal{}))
+}
+
+func testBuilder(t *testing.T, when spec.G, it spec.S) {
+	var copyDir, containerName, cacheVolume string
+
+	it.Before(func() {
+		containerName = "test-container-" + h.RandString(10)
+		var err error
+		copyDir, err = ioutil.TempDir("", "test-docker-copy-")
+		h.AssertNil(t, err)
+	})
+
+	it.After(func() {
+		if h.DockerContainerExists(t, containerName) {
+			h.Run(t, exec.Command("docker", "rm", containerName))
+		}
+		if h.DockerVolumeExists(t, cacheVolume) {
+			h.DockerVolumeRemove(t, cacheVolume)
+		}
+		os.RemoveAll(copyDir)
+	})
+
+	// .../cmd/lifecycle/builder.go#Args
+	when("called with arguments", func() {
+		it("errors", func() {
+			command := exec.Command(
+				"docker",
+				"run",
+				"--rm",
+				"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+				builderImage,
+				"some-arg",
+			)
+			output, err := command.CombinedOutput()
+			h.AssertNotNil(t, err)
+			expected := "failed to parse arguments: received unexpected arguments"
+			h.AssertStringContains(t, string(output), expected)
+		})
+	})
+
+	// .../cmd/lifecycle/builder.go#Privileges
+	when("running as a root", func() {
+		it("errors", func() {
+			command := exec.Command(
+				"docker",
+				"run",
+				"--rm",
+				"--user",
+				"root",
+				"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+				builderImage,
+			)
+			output, err := command.CombinedOutput()
+			h.AssertNotNil(t, err)
+			expected := "failed to build: refusing to run as root"
+			h.AssertStringContains(t, string(output), expected)
+		})
+	})
+
+	when("correct and full group.toml and plan.toml", func() {
+		it("succeeds", func() {
+			h.DockerRunAndCopy(t,
+				containerName,
+				copyDir,
+				ctrPath("/layers"),
+				builderImage,
+				h.WithFlags(
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+				),
+			)
+			// check builder metadata.toml for success test
+			md := getBuilderMetadata(t, filepath.Join(copyDir, "layers/config/metadata.toml"))
+
+			h.AssertStringContains(t, md.Buildpacks[0].API, "0.2")
+			h.AssertStringContains(t, md.Buildpacks[0].ID, "hello_world")
+			h.AssertStringContains(t, md.Buildpacks[0].Version, "0.0.1")
+		})
+	})
+
+	when("invalid input files", func() {
+		// .../cmd/lifecycle/builder.go#readData
+		when("group.toml", func() {
+			when("not found", func() {
+				it("errors", func() {
+					command := exec.Command(
+						"docker",
+						"run",
+						"--rm",
+						"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+						"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+						builderImage,
+					)
+					output, err := command.CombinedOutput()
+					h.AssertNotNil(t, err)
+					expected := "failed to read buildpack group"
```

Comment:
I think it's better to change the `expected` string in all of the tests to the full error.
For example, here it'll be:
```
failed to read buildpack group: open /layers/group.toml: no such file or directory
```
On line 174, it'll be:
```
failed to read buildpack group: Near line 1 (last key parsed 'I'): expected key separator '=', but got 'a' instead
```
(you can see more examples in `acceptance/detector_test.go` and `buildpack/detect_test.go`).

In this way, if something changes in the code that influence our tests, we'll be able to catch it more easily.

Reply:
fixed in 7e4d217